bigquery_etl/cli/glean_usage.py

```diff
@@ -104,19 +104,37 @@ def generate(project_id, output_dir, parallelism, exclude, only, app_name):
 
     app_info = [info for name, info in app_info.items() if name not in SKIP_APPS]
 
-    for table in GLEAN_TABLES:
-        with ThreadPool(parallelism) as pool:
-            pool.map(
-                partial(
-                    table.generate_per_app_id,
-                    project_id,
-                    output_dir=output_dir,
-                ),
-                baseline_tables,
-            )
-
-        with ThreadPool(parallelism) as pool:
-            pool.map(
-                partial(table.generate_per_app, project_id, output_dir=output_dir),
-                app_info,
-            )
+    # Prepare parameters so that generation of all Glean datasets can be done in parallel
+
+    # Parameters to generate per-app_id datasets consist of the function to be called
+    # and baseline tables
+    generate_per_app_id = [
+        (
+            partial(
+                table.generate_per_app_id,
+                project_id,
+                output_dir=output_dir,
+            ),
+            baseline_table,
+        )
+        for baseline_table in baseline_tables
+        for table in GLEAN_TABLES
+    ]
+
+    # Parameters to generate per-app_id datasets consist of the function to be called
+    # and app_info
+    generate_per_app = [
+        (
+            partial(table.generate_per_app, project_id, output_dir=output_dir),
+            app_info[0],
+        )
+        for table in GLEAN_TABLES
+    ]
+
+    with Pool(parallelism) as pool:
+        pool.starmap(run_generate, generate_per_app_id + generate_per_app)
```

Comment:
It's more efficient running all the tasks in parallel instead of iterating over the Glean tables and running the generation for each Glean table in parallel and waiting for it to finish every time.